dotnet/ios/ios12/ARKitVision/ARKitVision/ViewController.cs

```diff
@@ -0,0 +1,337 @@
+                        currentBuffer = frame.CapturedImage;
+                        ClassifyCurrentImage ();
+                }
+
+                frame.Dispose ();
+        }
+
+        #endregion
+
+        #region Vision classification
+
+        // Queue for dispatching vision classification requests
+        private readonly DispatchQueue visionQueue = new DispatchQueue ("com.example.apple-samplecode.ARKitVision.serialVisionQueue");
+
+        // The pixel buffer being held for analysis; used to serialize Vision requests.
+        private CVPixelBuffer? currentBuffer;
+
+        /// <summary>
+        /// Vision classification request and model
+        /// </summary>
+        private VNCoreMLRequest ClassificationRequest ()
+        {
+                var model = VNCoreMLModel.FromMLModel (new Inceptionv3 ().Model, out NSError error);
+                if (error is null)
+                {
+                        var request = new VNCoreMLRequest (model!, (internalRequest, internalError) =>
+                         {
+                                 ProcessClassifications (internalRequest, internalError);
+                         });
+
+                        // Crop input images to square area at center, matching the way the ML model was trained.
+                        request.ImageCropAndScaleOption = VNImageCropAndScaleOption.CenterCrop;
+
+                        // Use CPU for Vision processing to ensure that there are adequate GPU resources for rendering.
+                        request.UsesCpuOnly = true;
+
+                        return request;
+                }
+                else
+                {
+                        throw new Exception ($"Failed to load Vision ML model: {error}");
+                }
+        }
+
+        private void ClassifyCurrentImage ()
+        {
+                // Most computer vision tasks are not rotation agnostic so it is important to pass in the orientation of the image with respect to device.
+                var orientation = CGImagePropertyOrientationExtensions.ConvertFrom (UIDevice.CurrentDevice.Orientation);
+
+                var requestHandler = new VNImageRequestHandler (currentBuffer!, orientation, new VNImageOptions ());
+                visionQueue.DispatchAsync (() =>
+                 {
+                         requestHandler.Perform (new VNRequest [] { ClassificationRequest () }, out NSError error);
+                         if (error != null)
+                         {
+                                 Console.WriteLine ($"Error: Vision request failed with error \"{error}\"");
+                         }
+
+                         // Release the pixel buffer when done, allowing the next buffer to be processed.
+                         if (currentBuffer is not null)
+                         {
+                                 currentBuffer.Dispose ();
+                                 currentBuffer = null;
+                         }
```

Comment:
This is really a style choice, but another alternative is:

```suggestion
                         currentBuffer?.Dispose ();
                         currentBuffer = null;
```